samza-core/src/main/java/org/apache/samza/zk/TestZkProcessorLatch.java

```diff
@@ -0,0 +1,239 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.zk;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import org.I0Itec.zkclient.ZkClient;
+import org.I0Itec.zkclient.ZkConnection;
+import org.I0Itec.zkclient.exception.ZkNodeExistsException;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.MapConfig;
+import org.apache.samza.coordinator.CoordinationServiceFactory;
+import org.apache.samza.coordinator.ProcessorLatch;
+import org.apache.samza.testUtils.EmbeddedZookeeper;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+
+public class TestZkProcessorLatch {
+  private static EmbeddedZookeeper zkServer = null;
+  private static final ZkKeyBuilder KEY_BUILDER = new ZkKeyBuilder("test");
+  private ZkClient zkClient = null;
+  private static final int SESSION_TIMEOUT_MS = 20000;
+  private static final int CONNECTION_TIMEOUT_MS = 10000;
+  private ZkUtils zkUtils;
+  private ExecutorService pool = Executors.newFixedThreadPool(3);
+  private String zkConnectionString;
+
+  @BeforeSuite
+  public static void setup() throws InterruptedException {
+    zkServer = new EmbeddedZookeeper();
+    zkServer.setup();
+  }
+
+  @BeforeTest
+  public void testSetup() {
+    try {
+      zkConnectionString = "localhost:" + zkServer.getPort();
+      zkClient = new ZkClient(
+          new ZkConnection(zkConnectionString, SESSION_TIMEOUT_MS),
+          CONNECTION_TIMEOUT_MS);
+    } catch (Exception e) {
+      Assert.fail("Client connection setup failed. Aborting tests..");
+    }
+    try {
+      zkClient.createPersistent(KEY_BUILDER.getRootPath(), true);
+    } catch (ZkNodeExistsException e) {
+      Assert.fail("failed to create zk root");
+    }
+
+    zkUtils = new ZkUtils(
+        "testProcessorId",
+        KEY_BUILDER,
+        zkClient,
+        SESSION_TIMEOUT_MS);
+
+    zkUtils.connect();
+
+  }
+
+
+  @AfterTest
+  public void testTeardown() {
+    zkUtils.close();
+    zkClient.close();
+  }
+
+  @AfterClass
+  public static void teardown() {
+    zkServer.teardown();
+  }
+
+  @Test
+  public void testSingleLatch() {
```

Comment:
Size is 1. so one countdown should be sufficient for await to return immediately.
I also added another test with creates a 1-size latch, but two threads, and only one of them calls countDown();